### PR TITLE
Inline Keyboard for Telegram Alert

### PR DESCRIPTION
**Pull request by BlackYuzia**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Hi there.
I add inline keyboard use (like in discord webhooks, check screenshots)
Hope this improves view of telegram alert messages. 

## Type of change

- Other
- Alert changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods) (**I try to do this**)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## My Own Checklist

- [x] Test telegram alert (when service UP or DOWN)
- [x] Test telegram alert (when test alert)

## Screenshots 
image
image
I try get the same result as in discord webhooks
image

But, I found a bug (? probably) so maybe I should also duplicate title (name and status) to the body of telegram message? 🤔 
Empty message, so you can't see about what service you get alert:
image
But, that could be fixed by a few methods.

### CODE REVIEW

File: server/notification-providers/telegram.js [hunk 1]
```diff
@@ -1,18 +1,90 @@
...
+                    callback_data: "nothing",
+                }
+            ],
+            [
+                {
+                    text: heartbeatJSON?.time,
+                    callback_data: "nothing",
+                }
+            ]
+        ];
```
Comment (CommanderStorm) [inline server/notification-providers/telegram.js:43]: please refactor this code to be eazier readable through less code repetition

File: server/notification-providers/telegram.js [hunk 2]
```diff
@@ -1,18 +1,90 @@
...
+            // If heartbeatJSON is null, assume we're testing.
+            if (heartbeatJSON == null) {
+                await axios.get(`https://api.telegram.org/bot${notification.telegramBotToken}/sendMessage`, {
+                    params: {
+                        chat_id: notification.telegramChatID,
+                        text: msg,
+                    },
+                });
+                return okMsg;
+            }
```
Comment (CommanderStorm) [inline server/notification-providers/telegram.js:75]: If heartbeatJSON is null, this is a general message, not testing
```suggestion
```

File: server/notification-providers/telegram.js [hunk 3]
```diff
@@ -1,18 +1,90 @@
...
+            // https://core.telegram.org/bots/api#sendmessage
             await axios.get(`https://api.telegram.org/bot${notification.telegramBotToken}/sendMessage`, {
                 params: {
                     chat_id: notification.telegramChatID,
-                    text: msg,
+                    text: "­", // ALT + 0173 (that string isn't empty!)
+                    parse_mode: "markdown",
+                    reply_markup: {
+                        // https://core.telegram.org/bots/api#inlinekeyboardmarkup
+                        // https://core.telegram.org/bots/api#inlinekeyboardbutton
```
Comment (CommanderStorm) [inline server/notification-providers/telegram.js:85]: Move these up where the inlineKeyboard is defined, respectively.
Comments rot, these will particularly quickly.

File: server/notification-providers/telegram.js [hunk 4]
```diff
@@ -1,18 +1,90 @@
...
+                });
+                return okMsg;
+            }
+
+            // https://core.telegram.org/bots/api#sendmessage
             await axios.get(`https://api.telegram.org/bot${notification.telegramBotToken}/sendMessage`, {
                 params: {
                     chat_id: notification.telegramChatID,
-                    text: msg,
+                    text: "­", // ALT + 0173 (that string isn't empty!)
```
Comment (CommanderStorm) [inline server/notification-providers/telegram.js:81]: Please explain why this is nessesary/better

File: server/notification-providers/telegram.js [hunk 5]
```diff
@@ -1,18 +1,90 @@
...
+
+        // add error text to message
+        if (heartbeatJSON?.status === DOWN) {
+            inlineKeyboard.push([
+                {
+                    text: `Error: ${heartbeatJSON?.msg}`,
+                    callback_data: "nothing"
+                }
+            ]);
+        }
```
Comment (CommanderStorm) [inline server/notification-providers/telegram.js:53]: why is this in the inline keyboard and not in the text?

File: server/notification-providers/telegram.js [hunk 6]
```diff
@@ -1,18 +1,90 @@
...
+    /**
+     * @private
+     *
+     * @param {string} name service name
+     * @param {number} status 1 or 0
+     * @return {string}
+     * @memberof Telegram
+     */
+    getTitle(name, status) {
+        const emoji = status ? "✅" : "❌";
```
Comment (CommanderStorm) [inline server/notification-providers/telegram.js:18]: why is the emoji extracted?

File: server/notification-providers/telegram.js [hunk 7]
```diff
@@ -1,18 +1,90 @@
...
 
+    /**
+     * @private
+     *
+     * @param {string} name service name
+     * @param {number} status 1 or 0
+     * @return {string}
+     * @memberof Telegram
+     */
+    getTitle(name, status) {
```
Comment (CommanderStorm) [inline server/notification-providers/telegram.js:17]: why is this function extracted?

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
I have a few hard questions:
- why have you included information like `Ping` (which is not in discord embeds)
- Why is this better than a text message?
- Is it possible to replace only some part of the message with the inline keyboard?
  Imo only the last one benefits from the keyboard, the other ones could just as easily be in the text (or not there at all)

**Comment by louislam** [comment]
I checked the inline keywords is likely used for multiple choices or buttons. I feel that the info/message displaying on buttons is a bit strange to me.

https://core.telegram.org/bots/2-0-intro#new-inline-keyboards

**Pull request closed without merging**